### PR TITLE
[Patient Intake Sample] [Part 3] Add medical history and smoking status to the questionnaire

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Partial work #4999 - Add the following fields to the patient intake questionnaire:
  - Medical History
  - Family Member History
  - Smoking Status

## Steps to test

- [ ] Build the bots: `npm run build:bots`
- [ ] Upload core data
- [ ] Upload example bots
- [ ] Check if the changes are reflected in the questionnaire

[Screencast from 07-08-2024 17:46:49.webm](https://github.com/user-attachments/assets/e75e427e-77f0-48e5-970f-b0c837458c66)

[Screencast from 07-08-2024 18:39:38.webm](https://github.com/user-attachments/assets/25c60e48-08f6-429c-b6eb-a126ffd3afcc)

### CODE REVIEW

File: examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts [hunk 1]
```diff
@@ -116,14 +118,16 @@ export async function handler(medplum: MedplumClient, event: BotEvent<Questionna
...
+  if (answers['estimated-delivery-date']?.valueDate) {
+    await upsertObservation(
+      medplum,
+      patient,
+      observationCodeMapping.estimatedDeliveryDate,
+      observationCategoryMapping.socialHistory,
+      'valueDateTime',
+      { valueDateTime: convertDateToDateTime(answers['estimated-delivery-date'].valueDate) }
+    );
+  }
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/src/bots/core/intake-form.ts:130]: Drive-by fix.

File: examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json [hunk 2]
```diff
@@ -47,6 +47,62 @@
...
+    {
+      "fullUrl": "urn:uuid:183a569a-7f6d-41f9-97d8-57691fbc2876",
+      "request": {
+        "method": "PUT",
+        "url": "ValueSet?name=smoking-status-codes"
+      },
+      "resource": {
+        "resourceType": "ValueSet",
+        "status": "active",
+        "url": "http://example.com/smoking-status",
```
Comment (pamella) [inline examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:59]: I took a look at the [US Core Smoking Status Observation Profile](http://hl7.org/fhir/us/core/StructureDefinition/us-core-smokingstatus) and tried to use the `SmokingStatusComprehensive` ValueSet but I got a not found error. So I used the same options from the [SmokingStatus component](https://github.com/medplum/medplum/blob/3844220953b3ccde399dc52d5cb8390d361571d9/packages/react/src/PatientSummary/SmokingStatus.tsx#L14-L23).

image

Comment (rahul1) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:59]: @pamella  you might be looking at a later version of US Core. Right now, we implement [U.S. Core 5.0.0,](https://hl7.org/fhir/us/core/STU5/StructureDefinition-us-core-smokingstatus.html) which uses _this_ :

http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.11.20.9.38

Comment (rahul1) [reply examples/medplum-patient-intake-demo/data/core/patient-intake-questionnaire.json:59]: See my comment about the US State valuset (using `expansion`, version `0.0.0`) to create this valueset as well

**Review by rahul1 — APPROVED** [review]

**Comment by pamella** [comment]
@rahul1 Thanks! I'll fix that in the PR https://github.com/medplum/medplum/pull/5059
